skyportal/handlers/api/shift.py

```diff
@@ -623,89 +636,99 @@ def get(self, shift_id=None):
             if (end_date - start_date).days > 28:
                 return self.error("Please provide a period of less than 4 weeks")
 
-        report = {}
-        if start_date and end_date:
-            s = (
-                Shift.query_records_accessible_by(
-                    self.current_user,
-                    mode="read",
-                    options=[
-                        joinedload(Shift.shift_users),
-                    ],
+        with self.Session() as session:
+
+            report = {}
+            if start_date and end_date:
+                s = (
+                    session.scalars(
+                        Shift.select(
+                            session.user_or_token,
+                            mode="read",
```

Comment:
```suggestion
```

Reply:
Can just get rid of all these `mode="read"` in general.

Reply:
Yup will do.